components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -506,20 +516,52 @@ private boolean isAccountDisabled(String username, String tenantDomain) throws P
         return false;
     }
 
+    /**
+     * Check whether the user account is pending email verification.
+     *
+     * @param username     Username.
+     * @param tenantDomain Tenant domain.
+     * @return Whether user is pending email verification or not.
+     * @throws PostAuthenticationFailedException If error occurred while checking pending email verification status.
+     */
+    private boolean isPendingAccountVerification(String username, String tenantDomain)
+            throws PostAuthenticationFailedException {
+
+        try {
+            UserRealm realm = (UserRealm) FrameworkServiceDataHolder.getInstance().getRealmService()
+                    .getTenantUserRealm(IdentityTenantUtil.getTenantId(tenantDomain));
+            UserStoreManager userStoreManager = realm.getUserStoreManager();
+            Map<String, String> claimValues = userStoreManager.getUserClaimValues(username, new String[]{
+                    FrameworkConstants.ACCOUNT_STATE_CLAIM_URI}, UserCoreConstants.DEFAULT_PROFILE);
+            if (claimValues != null && !claimValues.isEmpty()) {
+                String accountStateClaim = claimValues.get(FrameworkConstants.ACCOUNT_STATE_CLAIM_URI);
+                return FrameworkConstants.AccountStatus.PENDING_LR.equals(accountStateClaim) ||
+                        FrameworkConstants.AccountStatus.PENDING_EV.equals(accountStateClaim) ||
+                        FrameworkConstants.AccountStatus.PENDING_SR.equals(accountStateClaim);
+            }
+        } catch (UserStoreException e) {
+            throw new PostAuthenticationFailedException(
+                    ErrorMessages.ERROR_WHILE_CHECKING_PENDING_VERIFICATION_STATUS.getCode(),
+                    String.format(ErrorMessages.ERROR_WHILE_CHECKING_PENDING_VERIFICATION_STATUS.getMessage(),
+                            username), e);
+        }
+        return false;
+    }
+
     private void handleAccountLockLoginFailure(String retryPage, AuthenticationContext context,
                                                HttpServletResponse response, String retryParam)
             throws PostAuthenticationFailedException {
 
         try {
-            // ToDo: Add support to configure enable/disable authentication failure reason.
-            boolean showAuthFailureReason = true;
             retryPage = FrameworkUtils.appendQueryParamsStringToUrl(retryPage,
                     "sp=" + context.getServiceProviderName());
             retryPage = FrameworkUtils.appendQueryParamsStringToUrl(retryPage,
                     String.format("%s=", FrameworkConstants.REQUEST_PARAM_AUTH_FLOW_ID)
                             + context.getContextIdentifier());
-            if (!showAuthFailureReason) {
-                retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
+            if (!doShowFailureReason()) {
+                retryParam =
+                        "&authFailure=true&authFailureMsg=login.failed.generic&errorCode=" +
```

Comment:
Why do we change the default message here ?

Reply:
With the introduction of the config to show auth failure reason, there needs to be a generic message which doesn't show the auth failiure reason when the config is turned off. This default message serves that purpose.